File: ch01/ch01_09.rb
def typoglycemia(text)
  result = []
  text.split.each do |word|
    if word.length < 5
      result.append(word)
    else
      result.append(word[0] + word[1..-2].chars.sample(word.length-2).join + word[-1])
    end
  end
  result.join(' ')
end

text = "I couldn't believe that I could actually understand what I was reading : the phenomenal power of the human mind ."
puts typoglycemia(text)

File: ch01/ch01_09.py
import random

def typoglycemia(text):
  result = []
  for word in text.split():
    if len(word) < 5:
      result.append(word)
    else:
      result.append(word[0] + ''.join(random.sample(word[1:-1], len(word)-2)) + word[-1])
  return ' '.join(result)

text = "I couldn't believe that I could actually understand what I was reading : the phenomenal power of the human mind ."
print(typoglycemia(text))
